python/private/parse_whl_name.bzl

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A starlark implementation of a Wheel distribution name
+"""
+
+def parse_whl_name(file):
+    """Parse whl file name into a struct of constituents.
+
+    TODO: add tests
+
+    Args:
+        file (str): The file name of a wheel
+
+    Returns:
+        a struct with the following attributes
+            distribution: the distribution name
+            version: the version of the distribution
+            build_tag: the build tag for the wheel
+            python_tag: the python tag for the wheel
+            abi_tag: the ABI tag for the wheel
+            platform_tag: the platform tag
+    """
+    if not file.endswith(".whl"):
+        fail("not a valid wheel: {}".format(file))
+
+    file = file[:-len(".whl")]
+
+    # Parse the following
```

Comment:
nit: probably link out to the PEPs somewhere.

https://packaging.python.org/en/latest/specifications/binary-distribution-format/#binary-distribution-format
https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/